software/python/src/harp_env_sensor/models/device.py

```diff
@@ -0,0 +1,115 @@
+import logging
+import os
+import serial.tools.list_ports as list_ports
+
+from pyharp.device import Device
+from pyharp.messages import MessageType, ReplyHarpMessage
+from typing import Dict, List, Optional, Union
+
+from ..models.register import Register
+
+
+class EnvSensorDevice(Device):
+    """
+    StepperMotorDevice class represents the StepperMotor pyharp DEVICE.
+    This device can control up to 4 separate stepper motors.
+    """
+
+    def __init__(self, com_port: Optional[str] = None, config: Dict = {}, dump_file: str = "ibl.bin"):
+        """
+        Initialize the StepperMotorDevice class
+
+        Stepper Motor Device contains 4 encoder, 4 inputs, and 4 stepper motors connected to it.
+        """
+        self.config = config
+
+        # Create directory for dump_file if it doesn't already exist
+        dump_file_path = config["dump_file_path"]
+        if not os.path.exists(dump_file_path):
+            os.makedirs(dump_file_path)
+
+        # Automatically read com ports if one is not given
+        if not com_port:
+            for port in list_ports.comports():
+                if port.pid == 24577:  # All harp devices have the same PID
+                    com_port = port.device
+                    super().__init__(com_port, dump_file_path + dump_file)
+                    if self.WHO_AM_I == 1234:  # Stepper Device ID
```

Comment:
This may have been copy-paste from elsewhere, but I think we want to change this to our environment sensor, probably extracted from the [yaml](https://github.com/AllenNeuralDynamics/harp.device.environment-sensor/blob/dev/device.yml#L5).